ignite/engine/__init__.py

```diff
@@ -91,20 +101,54 @@ def create_supervised_trainer(
 
     device_type = device.type if isinstance(device, torch.device) else device
     on_tpu = "xla" in device_type if device_type is not None else False
+    has_native_amp, has_apex_amp = False, False
 
     if on_tpu and not idist.has_xla_support:
         raise RuntimeError("In order to run on TPU, please install PyTorch XLA")
 
+    if amp and LooseVersion(torch.__version__) >= LooseVersion("1.6.0"):
```

Comment:
I think we can not do amp with TPU ? Maybe, we should explicitly raise an error if on_tpu is True and user passes amp and scaler...

Reply:
I totally forgot again :\